### ISSUE TITLE
Double encoding of comma in createCloudinaryURL()

### ISSUE THREAD

**Issue by Edjevw12** [cloudinary/js-url-gen#580]
In the createCloudinaryURL method, the comma , character in the publicID is being encoded twice.

This results in %2C being transformed into %252C, which causes issues when trying to access images with a comma in their publicID.

This issue is causing 404 errors when trying to access images with a comma in their publicID, as the double-encoded comma does not match the actual publicID of the image.

https://github.com/cloudinary/js-url-gen/blob/master/src/assets/CloudinaryFile.ts

the issue is in these lines of code

```js
const publicID = this.publicID
  .replace(/,/g, '%2C');

const safeURL = encodeURI(url)
  .replace(/\?/g, '%3F')
  .replace(/=/g, '%3D')
```

A fix could be to replace the double encoded characters so single encoded

```js
const safeURL = encodeURI(url)
  .replace(/\?/g, '%3F')
  .replace(/=/g, '%3D')
  .replace(/%252C/g, '%2C');
```

I've tested this fix in my application(s) with multiple characters, commas, and slashes, and it seems to work correctly.

**Comment by Edjevw12** [comment]
note: seems like this is expected behaviour. But it causes 404's in our application(s)? 

![image](https://github.com/cloudinary/js-url-gen/assets/37038043/e75f4a6b-b768-4433-b5e2-2b0ac0c0ace5)

**Comment by tommyg-cld** [comment]
@Edjevw12 thanks for raising this, it's a known issue and we will update you when it's fixed. 

For now, I would recommend adding the `.replaceAll('%252C', '%2C').replaceAll('%252C', '%2C')` fix to your code after you get the delivery URL.

**Comment by Edjevw12** [comment]
> @Edjevw12 thanks for raising this, it's a known issue and we will update you when it's fixed.
> 
> For now, I would recommend adding the `.replaceAll('%252C', '%2C').replaceAll('%252C', '%2C')` fix to your code after you get the delivery URL.

thanks!

**Comment by tommyg-cld** [comment]
@Edjevw12 this has been fixed per the latest version 1.12.1 so please update it to this and let us know if you get any issues.
